integration_test.go

```diff
@@ -589,3 +626,92 @@ func generateData(numRows int, sleepTime time.Duration, sendBadRow bool) error {
 
 	return nil
 }
+
+// Log entry template with all bigquery fields
+type log_entry_alltypes struct {
+	StringField     string  `json:"StringField"`
+	BytesField      []byte  `json:"BytesField"`
+	IntegerField    int64   `json:"IntegerField"`
+	FloatField      float64 `json:"FloatField"`
+	NumericField    string  `json:"NumericField"`
+	BigNumericField string  `json:"BigNumericField"`
+	BooleanField    bool    `json:"BooleanField"`
+	TimestampField  int64   `json:"TimestampField"`
+	DateField       int32   `json:"DateField"`
+	TimeField       string  `json:"TimeField"`
+	DateTimeField   string  `json:"DateTimeField"`
+	GeographyField  string  `json:"GeographyField"`
+	RecordField     struct {
+		SubField1 string `json:"SubField1"`
+		SubField2 string `json:"SubField2"`
+	} `json:"RecordField"`
+	RangeField struct {
+		Start int64 `json:"start"`
+		End   int64 `json:"end"`
+	} `json:"RangeField"`
+	JSONField string `json:"JSONField"`
+}
+
+// data generation function including all suported BQ fields
+func generateAllDataTypes(numRows int) error {
+	// open file
+	file, err := os.OpenFile(logFileName, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		return err
+	}
+	logger := log.New(file, "", 0)
+
+	// send json marshalled data
+	for i := 0; i < numRows; i++ {
+		jsonData := map[string]interface{}{
+			"name": "Jane Doe",
+			"age":  28,
+		}
+		jsonBytes, err := json.Marshal(jsonData)
+		if err != nil {
+			return err
+		}
+
+		curr := log_entry_alltypes{
+			StringField:     "hello world",
+			BytesField:      []byte("hello bytes"),
+			IntegerField:    123,
+			FloatField:      123.45,
+			NumericField:    "123.45",
+			BigNumericField: "123456789.123456789",
+			BooleanField:    true,
+			//milliseconds since unix epoch
+			TimestampField: 1721974126000000,
+			//days since unix epoch
+			DateField:      19929,
+			TimeField:      "12:34:56",
+			DateTimeField:  "2024-07-26 12:30:00.45",
+			GeographyField: "POINT(1 2)",
+			RecordField: struct {
+				SubField1 string `json:"SubField1"`
+				SubField2 string `json:"SubField2"`
+			}{
+				SubField1: "sub field value",
+				SubField2: "45.6",
+			},
+			//hardcoded civil-time encoded value
+			RangeField: struct {
+				Start int64 `json:"start"`
+				End   int64 `json:"end"`
+			}{
+				Start: 139830307704277524,
+				End:   139830307704277525,
+			},
+			JSONField: string(jsonBytes),
+		}
+		entry, err := json.Marshal(curr)
+		if err != nil {
+			return err
+		}
+		logger.Println(string(entry))
```

Comment:
Is this needed?

Reply:
Yes this is how we've been send the data